services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/config/ApplicationConfiguration.java

```diff
@@ -38,84 +52,145 @@
+    ManagerConfiguration managerConfiguration() {
+        return new ManagerConfiguration();
     }
 
     @Bean
-    Map<String, ServiceConfiguration> serviceConfigurations(final ConfigurationMap configurationMap) {
-        return configurationMap.getServices();
+    @ConfigurationProperties(prefix = "web")
+    public WebConfiguration webConfiguration() {
+        return new WebConfiguration();
     }
 
-    private File getServicesRoot() {
-        File parent = new File(".").getAbsoluteFile();
-        while (parent != null && !root.equals(parent.getName())) {
-            parent = parent.getParentFile();
+    @Bean
+    public ApplicationRunner managerApplicationRunner(final ManagerConfiguration managerConfiguration, final Function<String, ManagedService> serviceProducer) {
+        Runnable runner;
+        switch (managerConfiguration.getMode()) {
+            case RUN:
+                runner = new ScheduleRunner(managerConfiguration, serviceProducer);
+                break;
+            case SHUTDOWN:
+                runner = new ScheduleShutdown(managerConfiguration, serviceProducer);
+                break;
+            case LOGGERS:
+                runner = new LoggingBouncer(managerConfiguration, serviceProducer);
+                break;
+            case CONFIG:
+            default:
+                runner = new ConfigChecker(managerConfiguration);
+                break;
         }
-        return parent;
-    }
+        LOGGER.info("Constructed runner for {} mode: {}", managerConfiguration.getMode(), runner);
 
-    @Bean("runnerBuilders")
-    public Map<String, ProcessBuilder> runnerBuilders(final Map<String, ServiceConfiguration> runnerConfigs) {
-        return runnerConfigs.entrySet().stream()
-                .map(e -> {
-                    ServiceConfiguration config = e.getValue();
-                    ProcessBuilder builder = config.getProcessBuilder();
-                    builder.directory(getServicesRoot());
-                    return new SimpleEntry<>(e.getKey(), builder);
-                })
-                .collect(Collectors.toMap(SimpleEntry::getKey, SimpleEntry::getValue));
+        return args -> {
+            LOGGER.info("Running runner for manager: {}", runner);
+            runner.run();
+            System.exit(0);
+        };
     }
 
-    public static class ConfigurationMap {
+    @Bean("managedServiceProducer")
+    public Function<String, ManagedService> managedServiceProducer(final ManagedClient client, final WebConfiguration clientConfig) {
+        return serviceName -> {
+            Supplier<Collection<URI>> uriSupplier = () -> {
+                Collection<URI> clientUris = clientConfig.getClientUri(serviceName);
+                LOGGER.debug("Service {} has client uris {}", serviceName, clientUris);
+                return clientUris;
+            };
+            return new ManagedService(client, uriSupplier);
+        };
+    }
 
+    public static class ManagerConfiguration {
```

Comment:
It's a lot easier with the inner class for spring reasons, particularly because the ManagerConfiguration loads up some ServiceConfigurations which spring has to be aware of when converting the yaml to objects.
I'd prefer to make a comment saying "yes this could be a separate class, BUT it's easier as an inner"

Reply:
ok lets leave it as an inner class then